client/reader/tag-stream/controller.js

```diff
@@ -26,7 +27,8 @@ export const tagListing = ( context, next ) => {
 	);
 	const state = context.store.getState();
 	const tag = getReaderTagBySlug( state, tagSlug );
-	const tagTitle = tag?.title || titlecase( trim( context.params.tag ) ).replace( /[-_]/g, ' ' );
+	const tagTitle =
+		tag?.title || capitalPDangit( titlecase( trim( context.params.tag ) ).replace( /[-_]/g, ' ' ) );
```

Comment:
- Can we avoid titlecase and trim by updating capitalPDangit function?
- Not related to this PR but replacing - with an empty space seems like a quick hack. We should fix the root cause.

Reply:
> Can we avoid titlecase and trim by updating capitalPDangit function?

I don't think so, at least not for title case, as it seems common for folks to pass in full sentences to the function which would require sentence cast.

> but replacing - with an empty space seems like a quick hack. We should fix the root cause.

I think the cause is the tag not having a title and going to this fallback? But the fallback uses the slug afaict which is going to use -'s instead of spaces, so the replacement may make sense for the fallback case using slugs.

Reply:
Yeah these both seem to be needed for different reasons for now. But thanks for raising it!